packages/playwright/src/agents/generateAgents.ts

```diff
@@ -210,19 +210,19 @@ export async function initClaudeCodeRepo() {
 }
 
 const vscodeToolMap = new Map<string, string[]>([
-  ['ls', ['listDirectory', 'fileSearch']],
-  ['grep', ['textSearch']],
-  ['read', ['readFile']],
-  ['edit', ['editFiles']],
-  ['write', ['createFile', 'createDirectory']],
+  ['ls', ['search/listDirectory', 'search/fileSearch']],
+  ['grep', ['search/textSearch']],
+  ['read', ['search/readFile']],
+  ['edit', ['edit/editFiles']],
+  ['write', ['edit/createFile', 'edit/createDirectory']],
 ]);
-const vscodeToolsOrder = ['createFile', 'createDirectory', 'editFiles', 'fileSearch', 'textSearch', 'listDirectory', 'readFile'];
-const vscodeToolPrefix = 'test_'; // FIXME: Remove this once VSCode rolls fix for https://github.com/microsoft/vscode/issues/267811.
+const vscodeToolsOrder = ['edit/createFile', 'edit/createDirectory', 'edit/editFiles', 'search/fileSearch', 'search/textSearch', 'search/listDirectory', 'search/readFile'];
+const vscodeMcpName = 'playwright-test';
 function saveAsVSCodeChatmode(agent: Agent): string {
   function asVscodeTool(tool: string): string | string[] {
     const [first, second] = tool.split('/');
     if (second)
-      return second.startsWith('browser_') ? vscodeToolPrefix + second : second;
+      return second.startsWith('browser_') ? `${vscodeMcpName}/${second}` : second;
```

Comment:
I think this should be unconditional now, because all tools including `test_setup_page` and similar are prefixed.